pkg/meta/sql.go

```diff
@@ -1946,9 +1946,20 @@ func (m *dbMeta) doFindStaleSessions(limit int) ([]uint64, error) {
 
 func (m *dbMeta) doRefreshSession() {
 	_ = m.txn(func(ses *xorm.Session) error {
-		n, err := ses.Cols("Expire").Update(&session2{Expire: m.expireTime()}, &session2{Sid: m.sid})
-		if err == nil && n == 0 {
-			err = fmt.Errorf("no session found matching sid: %d", m.sid)
+		var err error
+		var ok bool
+		s := session2{Sid: m.sid}
+		if ok, err = ses.ForUpdate().Get(&s); ok {
```

Comment:
It's common to update it with a different value, so we should update it first and double check  it if n is 0 (uncommon)

Reply:
ok